sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -40,7 +41,7 @@ import org.apache.spark.sql.types.DataType
 case class UserDefinedFunction protected[sql] (
     f: AnyRef,
     dataType: DataType,
-    inputTypes: Option[Seq[DataType]]) {
+    inputTypes: Option[Seq[ScalaReflection.Schema]]) {
```

Comment:
This is a stable API. Are we able to make this change in 2.4 instead of 3.0?

Reply:
(BTW it was PR https://github.com/apache/spark/pull/22259 that was merged)

We can add back accessors, constructors, if it would make life easier for callers. But if this is protected, who are the callers of this code we're accommodating? maybe some hacky but important integration?

We'd have to rename `inputTypes` and then add back an accessor with the old type.